submission_runner.py

```diff
@@ -206,18 +211,17 @@ def train_once(
     workload.create_summary_writer(log_dir)
 
   with profiler.profile('Initializing optimizer'):
-    optimizer_state = init_optimizer_state(
-        workload,
-        model_params,
-        model_state,
-        hyperparameters,
-        opt_init_rng,
-    )
-
-  logging.info('Initializing metrics bundle')
+    optimizer_state = init_optimizer_state(workload,
+                                           model_params,
+                                           model_state,
+                                           hyperparameters,
+                                           opt_init_rng)
+
+  logging.info('Initializing metrics bundle.')
   if tokenizer_vocab_path:
     workload.init_metrics_bundle(tokenizer_vocab_path)
-  wandb.init()
+  if wandb is not None:
```

Comment:
When using PyTorch DDP, should wandb be initialized in all python processes? If not, we can add a `RANK == 0` condition here.

Reply:
Aha yep, added that.